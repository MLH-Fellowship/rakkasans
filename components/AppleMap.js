import React from "react";
import { View, StyleSheet } from "react-native";
import MapView, { PROVIDER_GOOGLE, Marker } from "react-native-maps"; // remove PROVIDER_GOOGLE import if not using Google Maps

export default ({ height }) => (
  <View style={[styles.container, { height: height }]}>
    <MapView
      provider={PROVIDER_GOOGLE} // remove if not using Google Maps
      style={styles.map}
      region={{
        latitude: 36.6634,
        longitude: -87.4774,
        latitudeDelta: 0.015,
        longitudeDelta: 0.0121,
      }}
      height="80%"
    >
      <Marker
        coordinate={{
          latitude: 36.6637,
          longitude: -87.4774,
        }}
        title={"Base"}
        pinColor="blue"
      />
    </MapView>
  </View>
);

const styles = StyleSheet.create({
  container: {
    ...StyleSheet.absoluteFillObject,
    justifyContent: "flex-start",
    alignItems: "center",
  },
  map: {
    ...StyleSheet.absoluteFillObject,
  },
});
