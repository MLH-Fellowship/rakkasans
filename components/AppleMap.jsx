import React from 'react';
import { View, StyleSheet } from 'react-native';
import MapView, { PROVIDER_GOOGLE, Marker } from 'react-native-maps'; // remove PROVIDER_GOOGLE import if not using Google Maps

const styles = StyleSheet.create({
  container: {
    ...StyleSheet.absoluteFillObject,
    justifyContent: 'flex-start',
    alignItems: 'center',
  },
  map: {
    ...StyleSheet.absoluteFillObject,
    borderWidth: 0.6,
    borderColor: '#bebebe',
  },
});

export default ({ height }) => (
  <View style={[styles.container, { height }]}>
    <MapView
      provider={PROVIDER_GOOGLE} // remove if not using Google Maps
      style={styles.map}
      region={{
        latitude: 36.6634,
        longitude: -87.4774,
        latitudeDelta: 0.015,
        longitudeDelta: 0.0121,
      }}
      height="80%"
    >
      <Marker
        coordinate={{
          latitude: 36.6637,
          longitude: -87.4774,
        }}
        title="Base"
        pinColor="blue"
      />
    </MapView>
  </View>
);
