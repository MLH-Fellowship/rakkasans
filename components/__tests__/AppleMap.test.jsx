import React from 'react';
import renderer from 'react-test-renderer';

import AppleMap from '../AppleMap';

jest.useFakeTimers();
jest.mock('react-native-maps', () => {
  const React = require.requireActual('react');
  const MapView = require.requireActual('react-native-maps');

  class MockCallout extends React.Component {
    render() {
      return React.createElement('Callout', this.props, this.props.children);
    }
  }

  class MockMarker extends React.Component {
    render() {
      return React.createElement('Marker', this.props, this.props.children);
    }
  }

  class MockMapView extends React.Component {
    render() {
      return React.createElement('MapView', this.props, this.props.children);
    }
  }

  MockCallout.propTypes = MapView.Callout.propTypes;
  MockMarker.propTypes = MapView.Marker.propTypes;
  MockMapView.propTypes = MapView.propTypes;
  MockMapView.Marker = MockMarker;
  MockMapView.Callout = MockCallout;
  return MockMapView;
});

it('renders correctly', () => {
  const tree = renderer.create(<AppleMap />).toJSON();
  expect(tree).toMatchSnapshot();
});
