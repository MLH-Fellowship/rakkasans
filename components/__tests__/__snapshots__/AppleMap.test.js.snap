// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "bottom": 0,
        "justifyContent": "flex-start",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      },
      Object {
        "height": undefined,
      },
    ]
  }
>
  <MapView
    height="80%"
    region={
      Object {
        "latitude": 36.6634,
        "latitudeDelta": 0.015,
        "longitude": -87.4774,
        "longitudeDelta": 0.0121,
      }
    }
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <Marker
      coordinate={
        Object {
          "latitude": 36.6637,
          "longitude": -87.4774,
        }
      }
      pinColor="blue"
      title="Base"
    />
  </MapView>
</View>
`;
